scripts/gha/trigger_workflow.py

```diff
@@ -0,0 +1,138 @@
+
+Usage:
+python trigger_workflow.py -w workflow_filename -t github_token [-b branch_name]
+       [-r git_repo_url] [-p <input1> <value1> -p <input2> <value2> ...]'
+       [-C curl_command]
+
+If -r is unspecified, uses the current repo.
+If -c is unspecified, uses the current HEAD.
+"""
+
+import argparse
+import json
+import os
+import re
+import subprocess
+import time
+import urllib.parse
+
+def main():
+  args = parse_cmdline_args()
+  if args.repo is None:
+      args.repo=subprocess.check_output(['git', 'config', '--get', 'remote.origin.url']).decode('utf-8').rstrip('\n').lower()
+      print('autodetected repo: %s' % args.repo)
+  if args.branch is None:
+      args.branch=subprocess.check_output(['git', 'rev-parse', '--abbrev-ref', 'HEAD']).decode('utf-8').rstrip('\n')
+      print('autodetected branch: %s' % args.branch)
+  if not args.repo.startswith('https://github.com/'):
+      print('Error, only https://github.com/ repositories are allowed.')
+      exit(2)
+  (repo_owner, repo_name) = re.match(r'https://github\.com/([^/]+)/([^/.]+)', args.repo).groups()
+
+  # POST /repos/{owner}/{repo}/actions/workflows/{workflow_id}/dispatches
+  request_url = 'https://api.github.com/repos/%s/%s/actions/workflows/%s/dispatches' % (repo_owner, repo_name, args.workflow)
+  json_params = {}
+  for param in args.param:
+      json_params[param[0]] = param[1]
+  json_text = '{"ref":%s,"inputs":%s}' % (json.dumps(args.branch), json.dumps(json_params))
+  if args.verbose or args.dryrun:
+    print('request_url: %s' % request_url)
+    print('request_body: %s' % json_text)
+  if args.dryrun:
+    return(0)
+  
+  print('Sending request to GitHub API...')
+  run_output = subprocess.check_output([args.curl,
+                                        '-s', '-o', '-', '-w', '\nHTTP status %{http_code}\n',
+                                        '-X', 'POST',
+                                        '-H', 'Accept: application/vnd.github.v3+json',
+                                        '-H', 'Authorization: token %s' % args.token,
+                                        request_url, '-d', json_text]
+                                      + ([] if not args.verbose else ['-v'])).decode('utf-8').rstrip('\n')
+  if args.verbose:
+    print(run_output)
+  if not re.search('HTTP status 2[0-9][0-9]$', run_output):
+    if not args.verbose:
+      print(run_output)
+    # Super quick and dirty way to get the message text since the appended status code means that
+    # the contents are not valid JSON.
+    error_message = re.search(r'"message": "([^"]+)"', run_output).group(1)
+    print('%sFailed to trigger workflow %s: %s' % (
+      '::error ::' if args.in_github_action else '', args.workflow, error_message))
+    return(-1)
+
+  print('Success!')
+  time.sleep(5)  # Give a few seconds for the job to become queued.
+  # Unfortunately, the GitHub REST API doesn't return the new workflow's run ID.
+  # Query the list of workflows to find the one we just added.
```

Comment:
:(

Reply:
Added a customizable sleep time, at least. Or maybe I should add a retry?

Reply:
Yeah, I'm so mad they just return an empty response. 😛